packages/ui-components/src/Input/BaseInput.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+  ChangeEvent,
+  CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import classNames from "classnames/bind";
+import styles from "./styles.module.scss";
+
+export interface InputProps<T = string | number> {
+  type?: T extends string ? "text" : "number";
+  initialValue?: T;
+  value?: T;
+  onChange?: (value: T) => void;
+  className?: string;
+  style?: CSSProperties;
+  autoComplete?: string;
+  placeholder?: string;
+  name?: string;
+  disabled?: boolean;
+  invalid?: boolean;
+  prefix?: React.ReactNode;
+}
+
+const cx = classNames.bind(styles);
+
+export const BaseInput: React.FC<InputProps> = ({
+  name,
+  type = "text",
+  autoComplete = "off",
+  className,
+  style,
+  value: outerValue = "",
+  initialValue = "",
+  placeholder,
+  onChange,
+  disabled = false,
+  invalid = false,
+}) => {
```

Comment:
Currently, `InputProps` doesn't allow to provide any unspecified props, so there is nothing left in props and the user cannot add anything else.
Do you suggest that we need to allow users to define all possible Input values, even if we don't handle them inside of component? For instance, `onFocus` or `onKeyDown` events?

Reply:
I’m a little torn between allowing unhandled props or specifying every prop possible. As you said, I don’t want to prevent users from adding additional event handlers or accessibility props or data props or any valid prop of an HTML input. But in a previous implementation I found it unruly to specify every possible prop when the base case of the component will only use a small portion of the props.

Having said that, I get the feeling that passing unhandled props are anathema to type safety. But I suppose my answer to that would be that HTML isn’t type safe. I welcome further discussion if you have suggestions.

Reply:
We could extend props with `React.InputHTMLAttributes<HTMLInputElement>` which is React type for allowed `<input />` props.
- we preserve type safety
- restrict users to provide only valid props

Reply:
That’s perfect.

Reply:
Done